pkg/cloudprovider/aws/apis/v1alpha1/provider.go

```diff
@@ -49,6 +49,55 @@ type AWS struct {
 	// Tags to be applied on ec2 resources like instances and launch templates.
 	// +optional
 	Tags map[string]string `json:"tags,omitempty"`
+	// MetadataOptions for the generated launch template of provisioned nodes.
+	//
+	// This specifies the exposure of the Instance Metadata Service to
+	// provisioned EC2 nodes. For more information,
+	// see Instance Metadata and User Data
+	// (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html)
+	// in the Amazon Elastic Compute Cloud User Guide.
+	//
+	// Refer to recommended, security best practices
+	// (https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node)
+	// for limiting exposure of Instance Metadata and User Data to pods.
+	// +optional
+	MetadataOptions *MetadataOptions `json:"metadataOptions,omitempty"`
+}
+
+// MetadataOptions contains parameters for specifying the exposure of the
+// Instance Metadata Service to provisioned EC2 nodes.
+type MetadataOptions struct {
+	// HttpEndpoint enables or disables the HTTP metadata endpoint on provisioned
+	// nodes. If the parameter is not specified, the default state is "enabled".
+	//
+	// If you specify a value of "disabled", instance metadata will not be accessible
+	// on the node.
+	// +optional
+	HttpEndpoint string `json:"httpEndpoint,omitempty"`
```

Comment:
generally, acronyms and initialisms in go should have the same case i.e. `HTTPEndpoint`. Although I realize the aws-sdk-go doesn't follow that :(    

https://github.com/golang/go/wiki/CodeReviewComments#initialisms